WPMedia/fusion-news-theme-blocks
# TMEDIA-457: Fixup Paywall Block and clean up usePaywall Hook

## Description
Finalize the paywall block component.  Fix bug with layer/video.  Reduce footprint of usePaywall to only necessary functionality, simplifying logic.

## Jira Ticket
- [TMEDIA-457](https://arcpublishing.atlassian.net/browse/TMEDIA-457)

## Acceptance Criteria
See Ticket.

## Test Steps
1. Checkout this branch `git checkout tmedia-457-paywall-block`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/subscriptions-block`
3. Ensure that You have the following two article pages.  If not, re-import pbdata:
![image](https://user-images.githubusercontent.com/2287238/138727536-ec22396b-1bcf-40fb-b21d-99bc7baebebd.png)
    * pb data: https://corecomponents.arcpublishing.com/pb-data/mongo/corecomponents-local.tar.gz
4. Check that you have the Paywall Block added and configured (this could potentially be placed inside a shared Fragment like http://localhost/pagebuilder/editor/curate?p=pNFbp320KN4TQr&v=vIMZut10KN4TQr )
![image](https://user-images.githubusercontent.com/2287238/139155388-a011ae65-bc43-41b2-982f-aa365acba51f.png)

![image](https://user-images.githubusercontent.com/2287238/139155159-2e9d5493-5d3e-449b-992c-73e47762cb14.png)
5. Log in using: http://localhost/pf/account/login/?_website=the-gazette
6. Open the original Kitchen Sink Article:
    * http://localhost/pf/kitchen-sink-article-jc/?_website=the-gazette
7. Open Brent's copy:
    * http://localhost/pf/ksa/?_website=the-gazette
8. The Paywall should appear.
9. Log out.
10. Clear cookies and localStorage for localhost.
    * 
![image](https://user-images.githubusercontent.com/2287238/139154880-a89bd294-6d9c-4245-a66f-c263ea989b07.png)
    * 
![image](https://user-images.githubusercontent.com/2287238/139154928-4ab2bc43-f6cb-4b6a-95ed-5cdc05b73ca4.png)
11. Open the original Kitchen Sink Article:
    * http://localhost/pf/kitchen-sink-article-jc/?_website=the-gazette
12. Open Brent's copy:
    * http://localhost/pf/ksa/?_website=the-gazette
13. The register wall should appear

## Effect Of Changes
### Before
No paywall would appear for going over the allocated views.

### After
A Regwall appears when going over the view count (currently 1)
![image](https://user-images.githubusercontent.com/2287238/139155524-da4545b2-2c5e-44e4-9146-ac400348c46e.png)

A Paywall appears when going over the view count (currently 2)
![image](https://user-images.githubusercontent.com/2287238/139155815-6bb3b64c-22aa-4996-b0e1-2bfd84e2ebab.png)

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
